kani-compiler/src/kani_queries/mod.rs

```diff
@@ -41,4 +44,40 @@ impl QueryDb {
     pub fn args(&self) -> &Arguments {
         self.args.as_ref().expect("Arguments have not been initialized")
     }
+
+    pub fn add_modifies_contract(&mut self, name: InternedString, contract: AssignsContract) {
+        assert!(
+            self.modifies_contracts.insert(name, contract).is_none(),
+            "Invariant broken, tried adding second modifies contracts to: {name}",
+        )
```

Comment:
I prefer not having functions with side effects inside assertions, except for tests. It's more readable to separate the action from the validation.

```suggestion
        let replaced = self.modifies_contracts.insert(name, contract);
        assert_eq!(replaced, None, "Tried adding second modifies contracts to: {name}")
```

Note: The suggestion might need tweaking.